### PR TITLE
fix(md-datepicker): style fix and applying placeholder

### PR DESCRIPTION
**Pull request by shivi956**
Styling Fix in md-datepicker and applying custom  placeholder to calendar without failing validation for the first time.

## Description

The box is always red for the first time , ( we need to add a default value of YYY-MM-DD (placeholder)) but that is also failing on validation .) 

Screenshots on the basis :
1. We don't need to send a value attribute to the calender
2. This is for the first time when the input box is not dirty , and the input box is dirty then red is acceptable.

## Related Issue

N/A
## Motivation and Context

Now we can set the placeholder values and the validation will not fail for the first time.
## How Has This Been Tested?

Tested Locally setting up the project

## Screenshots:
**Before** (If applicable):

image
**After**
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/menu-overlay/scss/menu-overlay.scss [hunk 1]
```diff
@@ -30,6 +30,11 @@
       &[data-show] {
         display: block;
       }
+
+      &[data-show-date-picker] {
+        display: block;
+        top: -15px !important;
```
Comment (parsoman) [inline web-components/src/components/menu-overlay/scss/menu-overlay.scss:36]: Will this work for responsive behaviour too? And can we do it without using important?

File: web-components/src/components/menu-overlay/MenuOverlay.ts [hunk 2]
```diff
@@ -76,6 +80,7 @@ export namespace MenuOverlay {
     @property({ type: Boolean, attribute: "show-arrow" }) showArrow = false;
     @property({ type: Boolean }) disabled = false;
     @property({ type: String }) placement: MenuOverlay.Placement = "bottom";
+    @property({ type: Boolean}) isDatePicker = false;
```
Comment (parsoman) [inline web-components/src/components/menu-overlay/MenuOverlay.ts:83]: Why we need to add property to menu overlay component which is generic component? We should not make changes based on datepicker inside menu overlay

File: web-components/src/components/menu-overlay/scss/menu-overlay.scss (hunk 1 above)
Comment (shivi956) [reply web-components/src/components/menu-overlay/scss/menu-overlay.scss]: changed this

File: web-components/src/components/menu-overlay/MenuOverlay.ts (hunk 2 above)
Comment (shivi956) [reply web-components/src/components/menu-overlay/MenuOverlay.ts]: Removed this

**Review by parsoman — APPROVED** [review]

**Pull request merged**